src/test/java/org/mule/tools/apikit/MainAppScaffolderWithExistingConfigApiSyncTest.java

```diff
@@ -36,15 +46,45 @@ public void reScaffold() throws Exception {
     ResourceLoader testScaffolderResourceLoader = new TestScaffolderResourceLoader(TEST_RESOURCES_APISYNC + "/v1");
     ApiReference apiReference = ApiReference.create(RAML_RESOURCE_URL_V1, testScaffolderResourceLoader);
 
-    ScaffoldingResult result = scaffoldApi(RuntimeEdition.CE, apiReference);
+    ScaffoldingResult result = scaffoldApi(RuntimeEdition.CE, apiReference, null, null, null);
     verifySuccessfulScaffolding(result, TEST_RESOURCES_APISYNC + "/v1/api.xml");
 
     List<MuleConfig> muleConfig = new ArrayList<>(result.getGeneratedConfigs());
-    ScaffoldingResult secondScaffoldingResult = scaffoldApi(RuntimeEdition.CE, apiReference, muleConfig);
+    ScaffoldingResult secondScaffoldingResult = scaffoldApi(RuntimeEdition.CE, apiReference, null, muleConfig, null);
 
     verifySuccessfulScaffolding(secondScaffoldingResult, TEST_RESOURCES_APISYNC + "/v1/api.xml");
   }
 
+  @Test
+  public void testRescaffoldSameVersionsGlobals() throws Exception {
+    ScaffoldingConfiguration.Builder configurationBuilder = ScaffoldingConfiguration.builder();
+    List<MuleConfig> muleConfigs = new ArrayList<>();
+    InputStream api =
+        new FileInputStream("src/test/resources/rescaffolding-apisync-version-with-global-config/pre-existing/api.xml");
+    InputStream global =
+        new FileInputStream("src/test/resources/rescaffolding-apisync-version-with-global-config/pre-existing/global.xml");
+    muleConfigs.add(MuleConfigBuilder.fromStream(api));
+    muleConfigs.add(MuleConfigBuilder.fromStream(global));
+    configurationBuilder.withMuleConfigurations(muleConfigs);
+    ScaffolderContext context = ScaffolderContextBuilder.builder().withRuntimeEdition(RuntimeEdition.EE).build();
+    MainAppScaffolder mainAppScaffolder = new MainAppScaffolder(context);
+    ObjectMapper mapper = new ObjectMapper();
+    InputStream scaffoldingConfigurationFile =
+        new FileInputStream("src/test/resources/rescaffolding-apisync-version-with-global-config/pre-existing/configuration.json");
+    ScaffoldingAccessories scaffoldingAccessories =
+        mapper.readValue(scaffoldingConfigurationFile, ScaffoldingAccessories.class);
+    configurationBuilder.withExternalCommonFile(scaffoldingAccessories.getExternalCommonFile());
+    configurationBuilder.withApiId(scaffoldingAccessories.getApiId());
+    configurationBuilder.withProperties(scaffoldingAccessories.getProperties());
+    configurationBuilder
```

Comment:
`configurationBuilder` is never used here but this is a good example of what I suggested for `scaffoldApi`. Keep this code and refactor `scaffoldApi` to take this as an argument